pkg/scaffold/helm/chart.go

```diff
@@ -15,40 +15,185 @@
 package helm
 
 import (
+	"fmt"
+	"io/ioutil"
 	"os"
 	"path/filepath"
+	"strings"
 
 	"github.com/operator-framework/operator-sdk/pkg/scaffold"
 
+	"github.com/iancoleman/strcase"
 	log "github.com/sirupsen/logrus"
 	"k8s.io/helm/pkg/chartutil"
+	"k8s.io/helm/pkg/downloader"
+	"k8s.io/helm/pkg/getter"
+	"k8s.io/helm/pkg/helm/environment"
+	"k8s.io/helm/pkg/helm/helmpath"
 	"k8s.io/helm/pkg/proto/hapi/chart"
+	"k8s.io/helm/pkg/repo"
 )
 
-// HelmChartsDir is the relative directory within an SDK project where Helm
-// charts are stored.
-const HelmChartsDir string = "helm-charts"
+const (
 
-// CreateChartForResource creates a new helm chart in the SDK project for the
-// provided resource.
-func CreateChartForResource(r *scaffold.Resource, projectDir string) (*chart.Chart, error) {
-	log.Infof("Create %s/%s/", HelmChartsDir, r.LowerKind)
+	// HelmChartsDir is the relative directory within an SDK project where Helm
+	// charts are stored.
+	HelmChartsDir string = "helm-charts"
+
+	// DefaultAPIVersion is the Kubernetes CRD API Version used for fetched
+	// charts when the --api-version flag is not specified
+	DefaultAPIVersion string = "charts.helm.k8s.io/v1alpha1"
+)
+
+type FetchChartOptions struct {
+	Chart   string
+	Repo    string
+	Version string
+}
+
+func CreateChart(apiVersion, kind string, opts FetchChartOptions, projectDir string) (*scaffold.Resource, *chart.Chart, error) {
+	chartsDir := filepath.Join(projectDir, HelmChartsDir)
+	err := os.MkdirAll(chartsDir, 0755)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var (
+		r *scaffold.Resource
+		c *chart.Chart
+	)
+
+	// If we don't have a helm chart reference, scaffold the default chart
+	// from Helm's default template. Otherwise, fetch it.
+	if len(opts.Chart) == 0 {
+		r, c, err = scaffoldChart(apiVersion, kind, chartsDir)
+	} else {
+		r, c, err = fetchChart(apiVersion, kind, opts, chartsDir)
+	}
+	if err != nil {
+		return nil, nil, err
+	}
+	log.Infof("Create %s/%s/", HelmChartsDir, c.GetMetadata().GetName())
```

Comment:
To me the log implies the user should create this.

Reply:
I don't disagree. However, this log matches other logs when scaffolding files in a new project, so I went with that to be consistent.

https://github.com/operator-framework/operator-sdk/blob/5f3c3a50fa121e6cae05b9afced1469bc9fe2b7c/pkg/scaffold/scaffold.go#L171

Reply:
Yep. I think it makes sense. It looks like "Created" would make sense here and in scaffold.go.